lib/common/grpc-service.js

```diff
@@ -446,26 +447,43 @@ GrpcService.createDeadline_ = function(timeout) {
 };
 
 /**
- * Checks for a grpc error code and extends the Error object with additional
+ * Checks for a grpc status code and extends the supplied object with additional
  * information.
  *
  * @private
  *
- * @param {error} err - The original request error.
- * @return {error|null}
+ * @param {object} obj - The object to be extended.
+ * @param {object} response - The grpc response.
+ * @return {object|null}
  */
-GrpcService.getError_ = function(err) {
-  if (err && GRPC_ERROR_CODE_TO_HTTP[err.code]) {
-    var defaultErrorDetails = GRPC_ERROR_CODE_TO_HTTP[err.code];
-    err.code = defaultErrorDetails.code;
-    err.message = err.message || defaultErrorDetails.message;
-    return err;
+GrpcService.extendGrpcResponse_ = function(obj, response) {
```

Comment:
I think you understood that piece of code differently as its author, but considering my position as a fellow maintainer, it wasn't obvious to me how that function was supposed to work. That's all I was going for... making contributions easier and not leaving bits of code in that aren't as obvious as they can be.